matron/src/weaver.c

```diff
@@ -2586,6 +2586,7 @@ int _system_glob(lua_State *l) {
     glob_status = glob(pattern, glob_flags, NULL, &g);
 
     if (glob_status != 0) {
+        fprintf(stderr, "glob failed for pattern (%s): %s\n", pattern, strerror(errno));
```

Comment:
Sorry. I should have been more precise in my statement about addressing the error at the source. That comment was made w.r.t. the `Mod.scan` function. Printing an error here presumes that all cases where the glob pattern fails are error cases which may not be the case. Further the weaver functions in general use `luaL_error(...)` to signal an error so that the caller can determine the outcome. If anything `_system_glob` should probably be using `luaL_error(...)` instead of simply returning an error code as the second value.

If you are feeling motivated I'd recommend using `luaL_error` else leave the code as is.